### PR TITLE
[WIP] Chef-15 bootstrap support

### PR DESCRIPTION
**Pull request by vsingh-msys**
### Description
In order to make the plugin compatible to the chef -15 we required to refactored and modify existing way to bootstrap the node using knife ec2.

As knife windows have been removed so we need to implement windows bootstrapping as well.
### Issues Resolved

https://github.com/chef/knife-ec2/issues/576

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by Vasu1105** [comment]
@vsingh-msys I was going through this PR and I felt we should not re declare the bootstrap options which are already there in the Chef core. We have to figure out what we are doing wrong here.

File: lib/chef/knife/ec2_server_create.rb [hunk 1]
```diff
@@ -615,14 +614,22 @@ def bootstrap_for(server, fqdn)
...
+        name_args = [fqdn]
 
         if config_value(:chef_node_name)
           config[:chef_node_name] = evaluate_node_name(config_value(:chef_node_name))
         else
           config[:chef_node_name] = server.id
         end
-        bootstrap_common_params(bootstrap)
+        bootstrap_common_params
+        register_client
```
Comment (Vasu1105) [inline lib/chef/knife/ec2_server_create.rb]: Looks like a duplication. This should be taken care by parent class.

File: lib/chef/knife/ec2_server_create.rb [hunk 2]
```diff
@@ -739,130 +574,68 @@ def s3_validation_key
...
+        config[:secret] = s3_secret || config_value(:secret)
 
         # If --chef-tag is provided then it will be set in chef as single value e.g. --chef-tag "myTag"
         # Otherwise if --tag-node-in-chef is provided then it will tag the chef in key=value pair of --tags option
         # e.g. --tags "key=value"
-        if locate_config_value(:chef_tag)
-          bootstrap.config[:tags] = locate_config_value(:chef_tag)
-        elsif locate_config_value(:tag_node_in_chef)
-          bootstrap.config[:tags] = config[:tags]
+        if config_value(:chef_tag)
```
Comment (Vasu1105) [inline lib/chef/knife/ec2_server_create.rb]: I think we should remove the option --tag-node-in-chef now https://github.com/chef/knife-ec2/pull/581/files#diff-e2993b8367db674f58a052036f00052aR792 as its already been deprecated. @btm what do you think ?

**Comment by btm** [comment]
Merged in #584

File: lib/chef/knife/ec2_server_create.rb (hunk 1 above)
Comment (btm) [reply lib/chef/knife/ec2_server_create.rb]: This duplication was fixed in #584.

File: lib/chef/knife/ec2_server_create.rb (hunk 2 above)
Comment (btm) [reply lib/chef/knife/ec2_server_create.rb]: @Vasu1105 @vsingh-msys yeah, we should drop the deprecated options. They've been deprecated over a year now. including `--security-group-ids`.

**Pull request closed without merging**